### PR TITLE
Make it possible to only allow PRs against a select set of target branches

### PR DESCRIPTION
**Pull request by rwestberg**
Hi all,

Please review this change that makes it possible to only allow PRs to target a select set of branches.

Best regards,
Robin

## Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

## Approvers
 * Erik Helin ([ehelin](@edvbld) - **Reviewer**) **Note!** Review applies to 7b261d73abbb0367ddc413e52e6f110da4facb43

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 1]
```diff
@@ -64,17 +62,19 @@ private CheckRun(CheckWorkItem workItem, PullRequest pr, PullRequestInstance prI
...
-                        List<Review> allReviews, List<Review> activeReviews, Set<String> labels, CensusInstance censusInstance, Map<String, String> blockingLabels,
-                        IssueProject issueProject) {
-        var run = new CheckRun(workItem, pr, prInstance, comments, allReviews, activeReviews, labels, censusInstance, blockingLabels, issueProject);
+                        List<Review> allReviews, List<Review> activeReviews, Set<String> labels, CensusInstance censusInstance) {
+        var run = new CheckRun(workItem, pr, prInstance, comments, allReviews, activeReviews, labels, censusInstance);
         run.checkStatus();
     }
 
+    private boolean checkTargetBranch() {
+        var matcher = workItem.bot.allowedTargetBranches().matcher(pr.targetRef());
```
Comment (edvbld) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java]: Should we use a pattern or should we simply just enumerate the protected branches? It seems potentially overkill to support a regex, it seems like an anti-pattern to have a repository with so many protected branches that they are cumbersome to enumerate, so you need a regex :smiley:

**Comment by rwestberg** [comment]
How about enumerating the valid targets in the error message?

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 2]
```diff
@@ -64,17 +62,29 @@ private CheckRun(CheckWorkItem workItem, PullRequest pr, PullRequestInstance prI
...
     static void execute(CheckWorkItem workItem, PullRequest pr, PullRequestInstance prInstance, List<Comment> comments,
-                        List<Review> allReviews, List<Review> activeReviews, Set<String> labels, CensusInstance censusInstance, Map<String, String> blockingLabels,
-                        IssueProject issueProject) {
-        var run = new CheckRun(workItem, pr, prInstance, comments, allReviews, activeReviews, labels, censusInstance, blockingLabels, issueProject);
+                        List<Review> allReviews, List<Review> activeReviews, Set<String> labels, CensusInstance censusInstance) {
+        var run = new CheckRun(workItem, pr, prInstance, comments, allReviews, activeReviews, labels, censusInstance);
         run.checkStatus();
     }
 
+    private boolean checkTargetBranch() {
```
Comment (edvbld) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java]:
```suggestion
    private boolean isTargetBranchAllowed() {
```

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 3]
```diff
@@ -114,6 +124,14 @@ private boolean checkCommitAuthor(List<Commit> commits) throws IOException {
     private List<String> botSpecificChecks() throws IOException {
         var ret = new ArrayList<String>();
 
+        if (!checkTargetBranch()) {
```
Comment (edvbld) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java]:
```suggestion
        if (!isTargetBranchAllowed()) {
```

Comment (rwestberg) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java]: Sounds good, will change!

**Pull request closed without merging**